antlr-kotlin-runtime/src/commonMain/kotlin/com/strumenta/antlrkotlin/runtime/StringLineReader.kt

```diff
@@ -0,0 +1,13 @@
+// Copyright 2017-present Strumenta and contributors, licensed under Apache 2.0.
+// Copyright 2024-present Strumenta and contributors, licensed under BSD 3-Clause.
+package com.strumenta.antlrkotlin.runtime
+
+public class StringLineReader(str: String) : LineReader {
+  private val lines = str.lineSequence().iterator()
+
+  override fun readLine(): String =
+    if (lines.hasNext()) lines.next() else throw NoSuchElementException()
```

Comment:
Could this be instead changed into this?
```
readLineOrNull ?: throw NoSuchElementException()
```

Reply:
Addressed!